tests/cassettes/test_export/test_export_api.yaml

```diff
@@ -0,0 +1,2940 @@
+        climate model intercomparison project, coordinated by PCMDI (Program For Climate
+        Model Diagnosis and Intercomparison) on behalf of the World Climate Research
+        Program (WCRP) and provided input for the Intergovernmental Panel on Climate
+        Change (IPCC) 5th Assessment Report (AR5).The CMIP5 archive is managed via
+        the Earth System Grid Federation, a globally distributed archive, with various
+        gateways with advanced faceted search capabilities provided by a number of
+        participating organisations. Full details are available from the PCMDI CMIP5
+        pages (see linked documentation on this record).", "stac_version": "1.0.0"}'
+    headers:
+      Access-Control-Allow-Headers:
+      - DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Content-Range,Range
+      Access-Control-Allow-Methods:
+      - GET, POST, OPTIONS
+      Access-Control-Allow-Origin:
+      - '*'
+      Access-Control-Expose-Headers:
+      - DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Content-Range,Range
+      Connection:
+      - keep-alive
+      Content-Length:
+      - '14471'
+      Content-Type:
+      - application/json
+      Date:
+      - Thu, 15 May 2025 02:10:12 GMT
+      Server:
+      - nginx/1.23.4
+      X-Content-Type-Options:
+      - nosniff
+      X-Robots-Tag:
+      - noindex, nofollow
+    status:
+      code: 200
+      message: OK
+- request:
+    body: null
+    headers:
+      Accept:
+      - '*/*'
+      Accept-Encoding:
+      - gzip, deflate
+      Connection:
+      - keep-alive
+      User-Agent:
+      - python-requests/2.32.3
+    method: GET
+    uri: https://hirondelle.crim.ca/stac/collections/c604ffb6d610adbb9a6b4787db7b8fd7
+  response:
+    body:
+      string: '{"id": "c604ffb6d610adbb9a6b4787db7b8fd7", "type": "Collection", "links":
+        [{"rel": "items", "type": "application/geo+json", "href": "https://hirondelle.crim.ca/stac/collections/c604ffb6d610adbb9a6b4787db7b8fd7/items"},
+        {"rel": "parent", "type": "application/json", "href": "https://hirondelle.crim.ca/stac/"},
+        {"rel": "root", "type": "application/json", "href": "https://hirondelle.crim.ca/stac/"},
+        {"rel": "self", "type": "application/json", "href": "https://hirondelle.crim.ca/stac/collections/c604ffb6d610adbb9a6b4787db7b8fd7"},
+        {"rel": "items", "href": "http://stac:8000/stac/collections/c604ffb6d610adbb9a6b4787db7b8fd7/items",
+        "type": "application/geo+json"}, {"rel": "items", "href": "http://stac:8000/stac/collections/c604ffb6d610adbb9a6b4787db7b8fd7/items",
+        "type": "application/geo+json"}, {"rel": "items", "href": "http://stac:8000/stac/collections/c604ffb6d610adbb9a6b4787db7b8fd7/items",
+        "type": "application/geo+json"}, {"rel": "items", "href": "http://stac:8000/stac/collections/c604ffb6d610adbb9a6b4787db7b8fd7/items",
```

Comment:
Still not patched. 😅 
The invalid URLs do not cause issues with the client?

Reply:
No because the client either looks for links with `"rel" = "item"` (not `"items"`) or it uses the `/search` endpoint.

The [`collections/{id}/items`](https://github.com/radiantearth/stac-api-spec/tree/v1.0.0/ogcapi-features#collection-items-collectionscollectioniditems) endpoint isn't used directly by `pystac-client` when looking for items for some reason.

Reply:
That explains it! I really was under the impression that the `ItemCollection` iterator was simply using that endpoint. Seems unnecessary to do a catalog wide search when listing all collections and their items one by one. Doesn't that cause unnecessary spatio-temporal filters/compares that inevitably returns everything?

Reply:
I think that depends on what the API in the backend does.

In my opinion, pystac-client makes some odd choices. I think there are definitely places that it could be improved.

Reply:
Yeah. Seems to be using the `Catalog` client's `ItemSearch` by default, since it's the most generic way that allows finding items across collections and even the collectionless items.

To take advantage of iterative optimizations mentioned in other comments that could resume/skip processed collections/items by ID, you would have to explicitly use `Catalog.get_collections` to yield `CollectionClient` and call `CollectionClient.get_items(recursive=True)`. Speed up is probably only noticeable if resuming a download midway of very large collections, or to retrieve only the few newly recently added items to the entire catalog without reloading all other JSONs.